cloudify_agent/worker.py

```diff
@@ -47,30 +47,31 @@ def __init__(self, *args, **kwargs):
         self._registry = kwargs.pop('registry')
         super(CloudifyOperationConsumer, self).__init__(*args, **kwargs)
 
-    def _print_task(self, ctx, action, status=None):
-        if ctx['type'] in ['workflow', 'hook']:
-            prefix = '{0} {1}'.format(action, ctx['type'])
-            suffix = ''
-        else:
-            prefix = '{0} operation'.format(action)
-            suffix = '\n\tNode ID: {0}'.format(ctx.get('node_id'))
-
-        if status:
-            suffix += '\n\tStatus: {0}'.format(status)
-
-        tenant_name = ctx.get('tenant', {}).get('name')
-        logger.info(
-            '\n\t{prefix} on queue `{queue}` on tenant `{tenant}`:\n'
-            '\tTask name: {name}\n'
-            '\tExecution ID: {execution_id}\n'
-            '\tWorkflow ID: {workflow_id}{suffix}\n'.format(
-                tenant=tenant_name,
-                prefix=prefix,
-                name=ctx['task_name'],
-                queue=ctx.get('task_target'),
-                execution_id=ctx.get('execution_id'),
-                workflow_id=ctx.get('workflow_id'),
-                suffix=suffix))
+    def _print_task(self, ctx, action, handler, status=None):
+        with state.current_ctx.push(handler.ctx):
+            if ctx['type'] in ['workflow', 'hook']:
+                prefix = '{0} {1}'.format(action, ctx['type'])
+                suffix = ''
+            else:
+                prefix = '{0} operation'.format(action)
+                suffix = '\n\tNode ID: {0}'.format(ctx.get('node_id'))
+
+            if status:
+                suffix += '\n\tStatus: {0}'.format(status)
+
+            tenant_name = ctx.get('tenant', {}).get('name')
+            logger.info(
```

Comment:
I'm wondering if it's worth changing this to use logger.info('%{thing}s', {'thing': 'something'}) approach so that the message isn't rendered if logging level is set to WARNING or above?